src/emailmessagelistmodel.cpp

```diff
@@ -362,6 +362,10 @@ void EmailMessageListModel::setSearch(const QString &search)
         }
 
         m_searchCanceled = false;
+        // All options are disabled, nothing to search
+        if (tempKey.isEmpty()) {
+            return;
```

Comment:
I would have thought that you should still update m_search and m_searchKey before returning?